krun/util.py

```diff
@@ -260,6 +260,22 @@ def check_and_parse_execution_results(stdout, stderr, rc, config,
             err_s += "Exception string: %s\n" % str(e)
         err_s += "return code: %d\n" % rc
         err_s += "stdout:\n%s\n%s\n%s\n\n" % (rule, stdout, rule)
+
+        # In instrumentation mode, stderr will have been written to disk for
+        # parsing. Read the last 50 lines back (it may be huge) so as to offer
+        # the user at-least something relating to the error.
+        if instrument:
+            from krun.vm_defs import INST_STDERR_FILE
+            num_stderr_lines = 0
+            with open(INST_STDERR_FILE) as fh:
+                num_stderr_lines += 1
+            stderr_lines = []
+            with open(INST_STDERR_FILE) as fh:
+                for line_num, line in enumerate(fh):
+                    if line_num >= num_stderr_lines - 50:
```

Comment:
We should make the "50" a constant. I'd also suggest making it more like 100 (it's not totally uncommon to have 50 stack frames).

Reply:
sure